src/Microsoft.Performance.SDK.Runtime/ColumnBuilding/Builders/CallbackInvokers/ModeBuilderCallbackInvoker.cs

```diff
@@ -0,0 +1,53 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System;
+using Microsoft.Performance.SDK.Processing;
+using Microsoft.Performance.SDK.Processing.ColumnBuilding;
+using Microsoft.Performance.SDK.Runtime.ColumnBuilding.Processors;
+using Microsoft.Performance.SDK.Runtime.ColumnVariants.TreeNodes;
+
+namespace Microsoft.Performance.SDK.Runtime.ColumnBuilding.Builders.CallbackInvokers;
+
+/// <summary>
+///     Responsible for invoking a callback that builds a single mode of a modal column variant.
+/// </summary>
+internal sealed class ModeBuilderCallbackInvoker
```

Comment:
nit: these implementation are so small, could we just make them structs? not a big deal at all if you don't want to, but seems like they're ideal candidates.